src/UnisonLocal/Page/HomePage.elm

```diff
@@ -1,42 +1,208 @@
+update appContext msg model =
+    case msg of
+        FetchProjectsFinished (Success projectNames) ->
+            ( { projects =
+                    projectNames
+                        |> List.map
+                            (\p ->
+                                ( ProjectName.toString p
+                                , ( p, [] )
+                                )
+                            )
+                        |> Dict.fromList
+              }
+            , projectNames
+                |> List.map
+                    (fetchProjectBranches FetchProjectBranchesFinished
+                        >> HttpApi.perform appContext.api
+                    )
+                |> Cmd.batch
+            )
+
+        FetchProjectBranchesFinished (Success ( projectName, branches )) ->
+            ( { model
+                | projects =
+                    model.projects
+                        |> Dict.insert
+                            (ProjectName.toString projectName)
+                            ( projectName, branches )
+              }
+            , Cmd.none
+            )
+
+        _ ->
+            ( model, Cmd.none )
+
+
+
+-- EFFECTS
+
+
+fetchProjects :
+    (WebData (List ProjectName) -> msg)
+    -> HttpApi.ApiRequest (List ProjectName) msg
+fetchProjects finishedMsg =
+    LocalApi.projects
+        |> HttpApi.toRequest decodeProjectList (RemoteData.fromResult >> finishedMsg)
+
+
+fetchProjectBranches :
+    (WebData ( ProjectName, List BranchSlug ) -> msg)
+    -> ProjectName
+    -> HttpApi.ApiRequest ( ProjectName, List BranchSlug ) msg
+fetchProjectBranches finishedMsg projectName =
+    let
+        decodeWithProjectName =
+            decodeBranchList
+                |> Decode.map (Tuple.pair projectName)
+    in
+    LocalApi.projectBranches projectName
+        |> HttpApi.toRequest decodeWithProjectName (RemoteData.fromResult >> finishedMsg)
+
+
+
+-- DECODE
+
+
+decodeProjectList : Decode.Decoder (List ProjectName)
+decodeProjectList =
+    Decode.list <|
+        Decode.field "projectName" ProjectName.decode
+
+
+decodeBranchList : Decode.Decoder (List BranchSlug)
+decodeBranchList =
+    let
+        branchSlugDecode =
+            Decode.map BranchRef.branchSlugFromString Decode.string
+                |> Decode.andThen (Util.decodeFailInvalid "Invalid BranchName")
+    in
+    Decode.list <|
+        Decode.field "branchName" branchSlugDecode
+
+
+
+-- VIEW
+
+
+viewProjectList : Projects -> List (Html Msg)
+viewProjectList projects =
+    let
+        branchTag projectName branchName =
+            BranchRef.projectBranchRef branchName
+                |> (\branchRef ->
+                        BranchRef.toTag branchRef
+                            |> Tag.withClick
+                                (Route.projectBranchRoot projectName branchRef Perspective.relativeRootPerspective
+                                    |> Route.toUrlString
+                                    |> Click.href
```

Comment:
There's a `Link` module to help with using routes for links.

Reply:
I'm not sure I understand how to use the module here. I can't find anything to do with routes (only external). Or did you mean to replace the `Tag` with `Link.view`?

Reply:
Oh I guess Link here doesn't have an internal section, so I can see how my comment didn't make a lot of sense.

I meant to say that instead of using `Route.projectBranchRoot` we should add a `Link.projectBranchRoot` instead that returns a `Click`:

```elm
-- Link.elm

projectBranchRoot : ProjectName -> BranchRef -> Click msg
projectBranchRoot projectName branchRef =
    let
        pers =
            Perspective.relativeRootPerspective
    in
    toClick (Route.projectBranchRoot projectName branchRef pers)
    
    
toClick : Route -> Click msg
toClick =
    Route.toUrlString >> Click.href
```

Reply:
Makes sense! I'll add it later today

Reply:
@hojberg Done!